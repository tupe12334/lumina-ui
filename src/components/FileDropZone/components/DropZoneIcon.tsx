import { UploadIcon } from '../infrastructure/UploadIcon'
import styles from '../FileDropZone.module.css'

interface DropZoneIconProps {
  icon?: string
}

export function DropZoneIcon({ icon }: DropZoneIconProps) {
  if (icon) {
    return <span className={styles.emojiIcon}>{icon}</span>
  }

  return <UploadIcon className={styles.icon} size={48} />
}