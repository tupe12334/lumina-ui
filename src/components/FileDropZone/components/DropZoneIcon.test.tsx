import { render, screen } from '@testing-library/react'
import { DropZoneIcon } from './DropZoneIcon'

describe('DropZoneIcon', () => {
  test('renders correctly with default icon', () => {
    const { container } = render(<DropZoneIcon />)
    expect(container.firstChild).toMatchSnapshot()
  })

  test('renders default upload icon when no icon prop provided', () => {
    render(<DropZoneIcon />)

    // Check for SVG upload icon
    const svgIcon = document.querySelector('svg')
    expect(svgIcon).toBeInTheDocument()
  })

  test('renders emoji when icon prop is provided', () => {
    render(<DropZoneIcon icon="📁" />)

    expect(screen.getByText('📁')).toBeInTheDocument()
  })

  test('does not render SVG when emoji is provided', () => {
    render(<DropZoneIcon icon="📸" />)

    expect(screen.getByText('📸')).toBeInTheDocument()

    // SVG should not be present
    const svgIcon = document.querySelector('svg')
    expect(svgIcon).not.toBeInTheDocument()
  })


  test('works with various emoji types', () => {
    const emojis = ['📁', '📸', '🎵', '📄', '🖼️', '🎬']

    emojis.forEach(emoji => {
      const { rerender } = render(<DropZoneIcon icon={emoji} />)
      expect(screen.getByText(emoji)).toBeInTheDocument()

      if (emoji !== emojis[0]) {
        rerender(<DropZoneIcon icon={emoji} />)
      }
    })
  })

  test('handles empty string icon prop', () => {
    render(<DropZoneIcon icon="" />)

    // Should render default SVG icon for empty string
    const svgIcon = document.querySelector('svg')
    expect(svgIcon).toBeInTheDocument()
  })
})