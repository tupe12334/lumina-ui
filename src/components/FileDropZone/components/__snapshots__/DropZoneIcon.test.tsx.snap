// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DropZoneIcon > renders correctly with default icon 1`] = `
<svg
  class="_icon_3af3de"
  fill="none"
  height="48"
  stroke="currentColor"
  stroke-linecap="round"
  stroke-linejoin="round"
  stroke-width="2"
  viewBox="0 0 24 24"
  width="48"
>
  <path
    class="_icon_3af3de"
    d="M21 15v4a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-4"
  />
  <polyline
    class="_icon_3af3de"
    points="7,10 12,5 17,10"
  />
  <line
    class="_icon_3af3de"
    x1="12"
    x2="12"
    y1="5"
    y2="15"
  />
</svg>
`;
